pkgs/by-name/en/encore/go.nix

```diff
@@ -0,0 +1,87 @@
+{ fetchFromGitHub
+, stdenv
+, lib
+, go
+, ...
+}:
+stdenv.mkDerivation rec {
+  pname = "go-encore";
+  version = "1.22";
+
+  src = fetchFromGitHub {
+    owner = "encoredev";
+    repo = "go";
+    rev = "encore-go${version}";
+    sha256 = "sha256-SSa3CoUS6JMfs6T1PMb3NK8UeJnpLlTIn46/3oCTnL8=";
+  };
+
+  buildInputs = [ go ];
+
+  patchPhase =
+    let
+      goSrc = fetchFromGitHub {
+        owner = "golang";
+        repo = "go";
+        rev = "release-branch.go${version}";
+        hash = "sha256-amASGhvBcW90dylwFRC2Uj4kOAOKCgWmFKhLnA9dOgg=";
+      };
+    in
+    ''
+      # delete submodule and replace with the go source
+      rm -rf go
+      cp -r ${goSrc} go
+      chmod -R u+rw go
+
+      cd go
+
+      for patch in ../patches/*.diff; do
+        patch --verbose -p1 --ignore-whitespace < "$patch"
+      done
+
+      cp -p -P -v -R ../overlay/* ./
+
+      cd ..
+    '';
+
+  installPhase =
+    let
+      goarch = platform:
+        {
+          "aarch64" = "arm64";
+          "arm" = "arm";
+          "armv5tel" = "arm";
+          "armv6l" = "arm";
+          "armv7l" = "arm";
+          "i686" = "386";
+          "mips" = "mips";
+          "mips64el" = "mips64le";
+          "mipsel" = "mipsle";
+          "powerpc64" = "ppc64";
+          "powerpc64le" = "ppc64le";
+          "riscv64" = "riscv64";
+          "s390x" = "s390x";
+          "x86_64" = "amd64";
+        }.${platform.parsed.cpu.name}
+          or (throw "Unsupported system: ${platform.parsed.cpu.name}");
+    in
+    ''
+      mkdir -p /tmp/.gobuild-cache
+      GOCACHE=/tmp/.gobuild-cache go run . \
+        --goos "${stdenv.targetPlatform.parsed.kernel.name}" \
+        --goarch "${goarch stdenv.targetPlatform}"
+
+      mkdir -p $out/{bin,share/go}
+
+      cp -r dist/${stdenv.targetPlatform.parsed.kernel.name}_${goarch stdenv.targetPlatform}/encore-go/* $out/share/go
+      ln -s $out/share/go/bin/go $out/bin
+      mv $out/bin/go $out/bin/go-encore
+    '';
+
+  meta = with lib; {
+    description = "Encore's rolling fork of Go with added automatic tracing & instrumentation";
+    homepage = "https://encore.dev";
+    maintainers = with maintainers; [ luisnquin ];
+    platforms = platforms.linux;
```

Comment:
❓ Is this really linux only?

Reply:
@afh I have only tested this package on Linux.

Reply:
According to https://encore.dev is should be possible to run on darwin, @luisnquin. Happy to help test on darwin if helpful…

Reply:
Of course, I'm currently delivering this package in [this flake](https://github.com/luisnquin/nixpkgs-extra) first.

Just clone it, run `nix build` and execute the following commands:

```shell
cd "$(mktemp -d --suffix="-temp-env")"
/path/to/.../result/bin/encore app create --example=url-shortener # specify project name
cd <project-name>
/path/to/.../result/bin/encore run

```

You will need to have `docker` available in your system.

Reply:
`/usr/bin/env NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1 nix build --impure github:luisnquin/nixpkgs-extra` currently fails with:
```
trying https://github.com/encoredev/go/archive/encore-go1.22.tar.gz
 …
curl: (22) The requested URL returned error: 404
```

What am I doing wrong?

Reply:
I'm not sure about it, it works in my system (i have cleaned the store)

![image](https://github.com/user-attachments/assets/f34e27c3-6dcb-4f6f-a2b3-3171f0027b03)

Reply:
I think I have an idea what is going wrong, please allow for a few moments to verify.